Sources/TSCBasic/FileSystem.swift

```diff
@@ -168,6 +168,10 @@ public protocol FileSystem: AnyObject {
     /// Check whether the given path is accessible and writable.
     func isWritable(_ path: AbsolutePath) -> Bool
 
+    /// Returns `true` if a given path has a quarantine attribute applied if when file system supports this attribute.
+    /// Returns `false` if such attribute is not applied or it isn't supported.
+    func hasQuarantineAttribute(_ path: AbsolutePath) -> Bool
```

Comment:
Does it make sense to spell this as `hasAttribute(_:)`?  That would allow checking for extended attributes Linux.

Reply:
I previously discussed this with @tomerd and @abertelrud and we ended up with this specialized version, but I wonder if making it slightly more generic in terms of supporting other attributes also makes sense?